src/Symfony/Component/Console/Application.php

```diff
@@ -899,13 +899,13 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
         try {
             $exitCode = $command->run($input, $output);
         } catch (\Exception $e) {
-            $event = new ConsoleTerminateEvent($command, $input, $output, $e->getCode());
-            $this->dispatcher->dispatch(ConsoleEvents::TERMINATE, $event);
+            $terminateEvent = new ConsoleTerminateEvent($command, $input, $output, $e->getCode());
+            $exceptionEvent = new ConsoleExceptionEvent($command, $input, $output, $e, $terminateEvent->getExitCode());
```

Comment:
yes the event listeners can alter it (which is precisely the reason why the second event was reading it from the first event). There is a setter in the event class